libcontainer/container_linux.go

```diff
@@ -431,6 +435,81 @@ func (c *Container) signal(s os.Signal) error {
 	return nil
 }
 
+func (c *Container) killViaPidfd() error {
+	pidfd, err := unix.PidfdOpen(c.initProcess.pid(), 0)
```

Comment:
I know the old code was killing only the init process. But would it make sense to kill all the processes in the cgroup instead?

We can do it as another PR, if that makes sense.

Reply:
As you mentioned in the above, we can consider this order:
1. Use cgroup.kill if the kernel supports it
2. Use pidfd if the kernel supports it
3. Just send a signal otherwise

But I think we still need to consider whether the container has a private pid ns or not.
What I think it's that, it's reasonable to kill only the init process for a private pid ns container.

Reply:
Taking into account this: https://github.com/opencontainers/runc/pull/4517#discussion_r2026922989.

It seems simpler to kill pid1 if it has it's own pidns. Otherwise, cgroup.kill or, if not supported, send a signal.

Does it make sense?

Reply:
I don't think it's no need to use `cgroup.kill` to kill all process in the cgroup to kill the pid1 if it has a private own pidns. We just only need to kill the exact pid1 process.

Reply:
Exactly what I said in the last comment :)